documentation/svfsiplus/theory/fluid/readme.md

```diff
@@ -0,0 +1,258 @@
+$$
+
+This is the VMS-stabilized weak form for the Navier-Stokes-Brinkman equations <a href="#ref-7">[7]</a> <a href="#ref-5">[5]</a> <a href="#ref-6">[6]</a>. The first seven terms on the left-hand side correspond to the standard Galerkin weak form. The last five terms are the stabilization terms obtained via VMS. In deriving this equation, we used the continuity equation to obtain $w_{i}u_{k}u_{i,k} = w_{i}\left(u_{k}u_{i}\right)_{,k}$. We also applied the following assumptions <a href="#ref-5">[5]</a>,
+<li> $\frac{du'}{dt} = 0$,
+</li>
+<li> $u' = 0$ on $\Gamma_{g}$ and $\Gamma_{h}$,
+</li>
+<li> $\nabla^{s}\boldsymbol{w}:2\mu\nabla^{s}\boldsymbol{u}' = 0$.
+</li>
+
+We then add an additional stabilization term, $\int \frac{\bar{\tau}\tau_{SUPS}^{2}}{\rho} w_{i,k}r_{Mk}r_{Mj}u_{i,j} \,d\Omega$ <a href="#ref-3">[3]</a> <a href="#ref-4">[4]</a> <a href="#ref-2">[2]</a>, to obtain
+
+$$
+\int qu_{i,i} \,d\Omega + \int \rho w_{i}\frac{du_{i}}{dt} \,d\Omega + \int \rho w_{i}u_{k}u_{i, k} \,d\Omega + \int w_{i, j}\sigma_{ij} \,d\Omega + \int \frac{\mu}{K}w_{i}u_{i} \,d\Omega - \int w_{i}\rho b_{i} \,d\Omega - \int_{\Gamma_{h}} w_{i}h_{i} \,d\Gamma + \int \tau_{SUPS}\left(\frac{q_{,i}}{\rho} + w_{i,k}u_{k}\right)r_{Mi} \,d\Omega + \int \rho \nu_{LSIC}r_{C}w_{i,i} \,d\Omega - \int w_{i}\tau_{SUPS}r_{Mk}u_{i,k} \,d\Omega - \int w_{i,k}\frac{\tau_{SUPS}^{2}}{\rho}r_{Mi}r_{Mk} \,d\Omega - \int \frac{\nu}{K}w_{i}\tau_{SUPS}r_{Mi} \,d\Omega + \int \frac{\bar{\tau}\tau_{SUPS}^{2}}{\rho} w_{i,k}r_{Mk}r_{Mj}u_{i,j} \,d\Omega = 0.
+$$
+
+This equation is the full stabilized weak form used in fluid.cpp in svFSIplus.
+
+### Residuals
+
+We will temporally discretize the stabilized weak form using the generalized - $\alpha$ method. The resulting nonlinear equation will be linearized and solved iteratively using the Newton-Raphson (Newton) method.
+
+To compute the residuals for each element in the mesh, we separate the stabilized weak form into momentum and continuity components,
+
+$$
+\int \rho w_{i}\frac{du_{i}}{dt} \,d\Omega + \int \rho w_{i}u_{k}u_{i, k} \,d\Omega + \int w_{i, j}\sigma_{ij} \,d\Omega + \int \frac{\mu}{K}w_{i}u_{i} \,d\Omega - \int w_{i}\rho b_{i} \,d\Omega - \int_{\Gamma_{h}} w_{i}h_{i} \,d\Gamma + \int \tau_{SUPS}w_{i,k}u_{k}r_{Mi} \,d\Omega + \int \rho \nu_{LSIC}r_{C}w_{i,i} \,d\Omega - \int w_{i}\tau_{SUPS}r_{Mk}u_{i,k} \,d\Omega - \int w_{i,k}\frac{\tau_{SUPS}^{2}}{\rho}r_{Mi}r_{Mk} \,d\Omega - \int \frac{\nu}{K}w_{i}\tau_{SUPS}r_{Mi} \,d\Omega + \int \frac{\bar{\tau}\tau_{SUPS}^{2}}{\rho} w_{i,k}r_{Mk}r_{Mj}u_{i,j} \,d\Omega = 0,
+$$
+
+$$
+\int qu_{i,i} \,d\Omega + \int \tau_{SUPS}\frac{q_{,i}}{\rho}r_{Mi} \,d\Omega = 0.
+$$
+
+Then, by plugging weighting functions into these equations and holding the results true for any arbitrary $w_{ai}$ and $q_{a}$, we obtain the per-element momentum and continuity residuals,
```

Comment:
@JonathanPham I think these residuals are per-node and not per-element. It may be confusing saying per-element, since we solve per node.